app/src/main/java/com/example/lutsak/bitmapimage/MainActivity.java

```diff
@@ -0,0 +1,115 @@
+package com.example.lutsak.bitmapimage;
+
+import android.content.res.Resources;
+import android.graphics.Bitmap;
+import android.graphics.BitmapFactory;
+import android.os.AsyncTask;
+import android.support.v7.app.AppCompatActivity;
+import android.os.Bundle;
+import android.util.Log;
+import android.widget.ImageView;
+import java.io.IOException;
+import java.net.URL;
+
+public class MainActivity extends AppCompatActivity {
+
+    public static final String TAG = MainActivity.class.getSimpleName();
+    public static final String IMAGE_URL = "https://www.alcopa-auction.fr/assets/img/brand-volkswagen.png";
+    private static final int HALF_SCREEN_WIDTH = Resources.getSystem().getDisplayMetrics().widthPixels / 2;
+    private static final int HALF_SCREEN_HEIGHT = Resources.getSystem().getDisplayMetrics().heightPixels / 2;
+
+    private ImageView mImageView;
+    private DownLoadImageTask mLoadImageTask;
+
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_main);
+        Log.d(TAG, ">> onCreate");
+
+        mImageView = findViewById(R.id.image_view);
+
+        /*
+          Download the logo from online and set it as
+          ImageView image programmatically.
+         */
+        mLoadImageTask = (DownLoadImageTask) getLastCustomNonConfigurationInstance();
+        if (mLoadImageTask == null) {
+            Log.d(TAG, "create");
+            mLoadImageTask = new DownLoadImageTask();
+            mLoadImageTask.execute(IMAGE_URL);
+        }
+        mLoadImageTask.link(this);
+
+        Log.d(TAG, "<< onCreate ");
+    }
+
+    public Object onRetainCustomNonConfigurationInstance() {
```

Comment:
Ok. And now enable Do not keep activities (DNKA) and 
- start app
- wait untill the image downloads
- put application to background
- put application to foreground - you will see that image will be downloaded once more